From ee025936469de666ba0b9308c4955ca85a1ce251 Mon Sep 17 00:00:00 2001
From: Courteney van den Berg <vdberg@ffm.tc.iot.dtag.de>
Date: Mon, 31 Mar 2008 17:11:29 +0200
Subject: [PATCH] Fix library search path to include /lib and /usr/lib

---
 aclocal.m4 |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/aclocal.m4 b/aclocal.m4
index 85fd9db..691752b 100644
--- a/aclocal.m4
+++ b/aclocal.m4
@@ -2141,8 +2141,8 @@ linux* | k*bsd*-gnu)
   # Some rework will be needed to allow for fast_install
   # before this can be enabled.
   hardcode_into_libs=yes
-  sys_lib_search_path_spec="/usr/lib${libsuff} /lib${libsuff} /usr/local/lib${libsuff}"
-  sys_lib_dlsearch_path_spec="/usr/lib${libsuff} /lib${libsuff}"
+  sys_lib_search_path_spec="/usr/lib /usr/lib${libsuff} /lib /lib${libsuff} /usr/local/lib${libsuff}"
+  sys_lib_dlsearch_path_spec="/usr/lib /usr/lib${libsuff} /lib /lib${libsuff}"
 
   # Append ld.so.conf contents to the search path
   if test -f /etc/ld.so.conf; then
