From a33c3ef391c9bc6d164633e70cb8bb56ed738b06 Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd@debian.org>
Date: Sun, 19 Oct 2008 13:39:50 +0100
Subject: [PATCH] load module gconf earlier

---
 src/daemon/default.pa.in |   18 +++++++++---------
 1 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/src/daemon/default.pa.in b/src/daemon/default.pa.in
index 517f2c9..cdb62dd 100755
--- a/src/daemon/default.pa.in
+++ b/src/daemon/default.pa.in
@@ -76,6 +76,15 @@ load-module module-native-protocol-unix
 #load-module module-null-sink sink_name=rtp format=s16be channels=2 rate=44100 description="RTP Multicast Sink"
 #load-module module-rtp-send source=rtp.monitor
 
+### Load additional modules from GConf settings. This can be configured with the paprefs tool.
+### Please keep in mind that the modules configured by paprefs might conflict with manually
+### loaded modules.
+.ifexists module-gconf@PA_SOEXT@
+.nofail
+load-module module-gconf
+.fail
+.endif
+
 ### Automatically restore the default sink/source when changed by the user during runtime
 load-module module-default-device-restore
 
@@ -96,15 +105,6 @@ load-module module-console-kit
 ### Enable positioned event sounds
 load-module module-position-event-sounds
 
-### Load additional modules from GConf settings. This can be configured with the paprefs tool.
-### Please keep in mind that the modules configured by paprefs might conflict with manually
-### loaded modules.
-.ifexists module-gconf@PA_SOEXT@
-.nofail
-load-module module-gconf
-.fail
-.endif
-
 # X11 modules should not be started from default.pa so that one daemon
 # can be shared by multiple sessions.
 
