From dc6555053b5449c4bf079bce89fb6deb67a0634e Mon Sep 17 00:00:00 2001
From: Petr Salinger <Petr.Salinger@seznam.cz>
Date: Sat, 1 Sep 2007 23:19:37 +0200
Subject: [PATCH] Fix build on newer gcc versions

---
 src/pulsecore/core-util.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/pulsecore/core-util.c b/src/pulsecore/core-util.c
index 56dd693..7b7746f 100644
--- a/src/pulsecore/core-util.c
+++ b/src/pulsecore/core-util.c
@@ -544,6 +544,7 @@ fail:
         cap_free(caps);
     }
 #endif
+    ;
 }
 
 /* Reset the priority to normal, inverting the changes made by pa_raise_priority() */
-- 
1.5.3.rc7.36.g8feb9-dirty

