From c8c7a262a5a53ce1e8edbd5cca00b087d7e95d76 Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd@luon.net>
Date: Sat, 3 Nov 2007 21:13:38 +0100
Subject: [PATCH] Fix pa_readlink to put a \0 in the right location

---
 src/pulsecore/core-util.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/pulsecore/core-util.c b/src/pulsecore/core-util.c
index b33241a..b2bc5f0 100644
--- a/src/pulsecore/core-util.c
+++ b/src/pulsecore/core-util.c
@@ -1523,7 +1523,7 @@ char *pa_readlink(const char *p) {
         }
 
         if ((size_t) n < l-1) {
-            c[l-1] = 0;
+            c[n] = 0;
             return c;
         }
 
