From 121bfc9f1916395e5216927ec3a9e0bbd742395a Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd@debian.org>
Date: Sun, 19 Oct 2008 13:39:50 +0100
Subject: [PATCH] load module gconf earlier

---
 src/daemon/default.pa.in |   19 ++++++++++---------
 1 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/src/daemon/default.pa.in b/src/daemon/default.pa.in
index 7032038..41322c6 100755
--- a/src/daemon/default.pa.in
+++ b/src/daemon/default.pa.in
@@ -72,6 +72,16 @@ load-module module-native-protocol-unix
 #load-module module-null-sink sink_name=rtp format=s16be channels=2 rate=44100 description="RTP Multicast Sink"
 #load-module module-rtp-send source=rtp.monitor
 
+### Load additional modules from GConf settings. This can be configured with the paprefs tool.
+### Please keep in mind that the modules configured by paprefs might conflict with manually
+### loaded modules.
+.ifexists module-gconf@PA_SOEXT@
+.nofail
+load-module module-gconf
+.fail
+.endif
+
+
 ### Automatically restore the volume of streams and devices
 load-module module-stream-restore
 load-module module-device-restore
@@ -96,15 +106,6 @@ load-module module-console-kit
 ### Enable positioned event sounds
 load-module module-position-event-sounds
 
-### Load additional modules from GConf settings. This can be configured with the paprefs tool.
-### Please keep in mind that the modules configured by paprefs might conflict with manually
-### loaded modules.
-.ifexists module-gconf@PA_SOEXT@
-.nofail
-load-module module-gconf
-.fail
-.endif
-
 # X11 modules should not be started from default.pa so that one daemon
 # can be shared by multiple sessions.
 
