From 546cf4b92fa968d9a5242921272ad09c26cda049 Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd@luon.net>
Date: Sun, 4 Nov 2007 13:30:57 +0100
Subject: [PATCH] Define __NR_eventfd on arm if it wasn't defined yet

---
 src/pulsecore/fdsem.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/src/pulsecore/fdsem.c b/src/pulsecore/fdsem.c
index 927bf00..59eec18 100644
--- a/src/pulsecore/fdsem.c
+++ b/src/pulsecore/fdsem.c
@@ -53,6 +53,10 @@
 #define __NR_eventfd 284
 #endif
 
+#if !defined(__NR_eventfd) && defined(__arm__)
+#define __NR_eventfd (__NR_SYSCALL_BASE+351)
+#endif
+
 #if !defined(SYS_eventfd) && defined(__NR_eventfd)
 #define SYS_eventfd __NR_eventfd
 #endif
