Index: pulseaudio-0.9.22/man/pulse-daemon.conf.5.xml.in
===================================================================
--- pulseaudio-0.9.22.orig/man/pulse-daemon.conf.5.xml.in	2010-12-17 20:40:29.000000000 -0500
+++ pulseaudio-0.9.22/man/pulse-daemon.conf.5.xml.in	2010-12-20 21:36:39.000000000 -0500
@@ -89,20 +89,21 @@
       <opt>src-zero-order-hold</opt>, <opt>src-linear</opt>,
       <opt>trivial</opt>, <opt>speex-float-N</opt>,
       <opt>speex-fixed-N</opt>, <opt>ffmpeg</opt>. See the
-      documentation of libsamplerate for an explanation for the
-      different src- methods. The method <opt>trivial</opt> is the most basic
-      algorithm implemented. If you're tight on CPU consider using
-      this. On the other hand it has the worst quality of them
-      all. The Speex resamplers take an integer quality setting in the
-      range 0..9 (bad...good). They exist in two flavours: <opt>fixed</opt> and
-      <opt>float</opt>. The former uses fixed point numbers, the latter relies on
-      floating point numbers. On most desktop CPUs the float point
-      resmampler is a lot faster, and it also offers slightly better
-      quality. See the output of <opt>dump-resample-methods</opt> for
-      a complete list of all available resamplers. Defaults to
-      <opt>speex-float-3</opt>. The <opt>--resample-method</opt>
-      command line option takes precedence. Note that some modules
-      overwrite or allow overwriting of the resampler to use.</p>
+      documentation of libsamplerate and speex for explanations of the
+      different src- and speex- methods, respectively. The method
+      <opt>trivial</opt> is the most basic algorithm implemented. If
+      you're tight on CPU consider using this. On the other hand it has
+      the worst quality of them all. The Speex resamplers take an
+      integer quality setting in the range 0..10 (bad...good). They
+      exist in two flavours: <opt>fixed</opt> and <opt>float</opt>. The former uses fixed point
+      numbers, the latter relies on floating point numbers. On most
+      desktop CPUs the float point resampler is a lot faster, and it
+      also offers slightly better quality. See the output of
+      <opt>dump-resample-methods</opt> for a complete list of all
+      available resamplers. Defaults to <opt>speex-float-3</opt>. The
+      <opt>--resample-method</opt> command line option takes precedence.
+      Note that some modules overwrite or allow overwriting of the
+      resampler to use.</p>
     </option>
 
     <option>
