From 7d213f4c7a5328190626b5efdccc45901408650d Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd@luon.net>
Date: Sun, 27 Apr 2008 21:39:08 +0200
Subject: [PATCH] Define PULSE_INTERNAL

---
 src/daemon/main.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/daemon/main.c b/src/daemon/main.c
index 1e3b358..1b6fe3f 100644
--- a/src/daemon/main.c
+++ b/src/daemon/main.c
@@ -677,6 +677,7 @@ int main(int argc, char *argv[]) {
 #endif
     }
 
+    set_env("PULSE_INTERNAL", "1");
     pa_assert_se(chdir("/") == 0);
     umask(0022);
 
