From 44a4f2b01915544994bdff805d4b0a163f3454d0 Mon Sep 17 00:00:00 2001
From: Sjoerd Simons <sjoerd@luon.net>
Date: Sat, 10 May 2008 21:59:57 +0200
Subject: [PATCH] Reduce RT prio logging severity

---
 src/daemon/main.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/daemon/main.c b/src/daemon/main.c
index 1b6fe3f..74d1350 100644
--- a/src/daemon/main.c
+++ b/src/daemon/main.c
@@ -449,9 +449,10 @@ int main(int argc, char *argv[]) {
             suid_root = real_root = FALSE;
 
             if (conf->high_priority || conf->realtime_scheduling)
-                pa_log_notice("Called SUID root and real-time/high-priority scheduling was requested in the configuration. However, we lack the necessary priviliges:\n"
+                pa_log_info("Called SUID root and real-time/high-priority scheduling was requested in the configuration. However, we lack the necessary priviliges:\n"
                               "We are not in group '"PA_REALTIME_GROUP"' and PolicyKit refuse to grant us priviliges. Dropping SUID again.\n"
-                              "For enabling real-time scheduling please acquire the appropriate PolicyKit priviliges, or become a member of '"PA_REALTIME_GROUP"', or increase the RLIMIT_NICE/RLIMIT_RTPRIO resource limits for this user.");
+                              "For enabling real-time scheduling please acquire the appropriate PolicyKit priviliges, or become a member of '"PA_REALTIME_GROUP"', or increase the RLIMIT_NICE/RLIMIT_RTPRIO resource limits for this user.\n"
+                              "Note that real-time/high-priority scheduling is NOT normally required. If you experience crackling or other sound anomalies, consider one or more of the above solutions.");
         }
 
     } else {
