Description: Avoid abort when poll descriptor is 0 (LP: #1092377)

Index: pulseaudio/src/modules/alsa/alsa-mixer.c
===================================================================
--- pulseaudio.orig/src/modules/alsa/alsa-mixer.c
+++ pulseaudio/src/modules/alsa/alsa-mixer.c
@@ -328,6 +328,9 @@ static void defer_cb(pa_mainloop_api *a,
     if (n < 0) {
         pa_log("snd_mixer_poll_descriptors_count() failed: %s", pa_alsa_strerror(n));
         return;
+    } else if (n == 0) {
+        pa_log("snd_mixer_poll_descriptors_count() equal 0");
+        return;
     }
     num_fds = (unsigned) n;
 
