From a15f3868f0b645fa07e1555a5c2812516af86b1e Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Sun, 19 Apr 2009 19:25:42 +0200
Subject: [PATCH 28/28] alsa: remove debug code

---
 src/modules/alsa/alsa-util.c |    2 --
 1 files changed, 0 insertions(+), 2 deletions(-)

diff --git a/src/modules/alsa/alsa-util.c b/src/modules/alsa/alsa-util.c
index 98710d4..aae38bc 100644
--- a/src/modules/alsa/alsa-util.c
+++ b/src/modules/alsa/alsa-util.c
@@ -1920,8 +1920,6 @@ const char* pa_alsa_strerror(int errnum) {
     if ((t = PA_STATIC_TLS_GET(cstrerror)))
         pa_xfree(t);
 
-    errnum = EINVAL;
-    PA_DEBUG_TRAP;
     original = snd_strerror(errnum);
 
     if (!original) {
