Subject: Fix PulseAudio failure on some non-English locales
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/pulseaudio/+bug/1445358
Author: Simos Xenitellis

diff --git a/src/pulsecore/core-util.h b/src/pulsecore/core-util.h
index dcdc40e..e6be901 100644
--- a/src/pulsecore/core-util.h
+++ b/src/pulsecore/core-util.h
@@ -90,7 +90,7 @@ int pa_parse_boolean(const char *s) PA_GCC_PURE;
 int pa_parse_volume(const char *s, pa_volume_t *volume);
 
 static inline const char *pa_yes_no(bool b) {
-    return b ? _("yes") : _("no");
+    return b ? "yes" : "no";
 }
 
 static inline const char *pa_strnull(const char *x) {
