Index: pulseaudio-5.0/src/modules/alsa/alsa-ucm.c
===================================================================
--- pulseaudio-5.0.orig/src/modules/alsa/alsa-ucm.c	2014-03-19 13:15:43.734768910 +1100
+++ pulseaudio-5.0/src/modules/alsa/alsa-ucm.c	2014-03-19 13:15:43.730768910 +1100
@@ -1541,7 +1541,8 @@
         ucm_create_profile(ucm, ps, verb, verb_name, verb_desc);
     }
 
-    ucm_probe_profile_set(ucm, ps);
+/*  Just trust that the person writing the UCM file knows what (s)he was doing, right? */
+/*    ucm_probe_profile_set(ucm, ps); */
     ps->probed = true;
 
     return ps;
